### PR TITLE
3469-1

### PR DESCRIPTION
**Pull request by Morozov-Alexander**
## Номер
3469

## Номер задания
1

## Ссылка на видео с демо
https://youtu.be/sJY0AmnWWRw

### CODE REVIEW

**Comment by DmitryAvramec** [comment]
image
Next time if you request a review and I will see this again you will understand why is it better join Marce's program this summer

**Comment by Morozov-Alexander** [comment]
Personally, I have not asked anyone for a review, at the moment I am only correcting my mistakes.

File: .reek.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+detectors:
+  IrresponsibleModule:
```
Comment (Vadser) [inline .reek.yml]: Don't touch files, which is not in your folder, return all such files in original state

File: 3469/1/api.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class Api
```
Comment (Vadser) [inline 3469/1/api.rb]: Please make the class name more convenient

File: 3469/1/api.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  attr_reader :gem_information
+  def initialize(key, name_of_gem)
```
Comment (Vadser) [inline 3469/1/api.rb]: It is bad practice to return value from initialize, you should add another public method

File: 3469/1/html_parser.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class HtmlParser
+  attr_reader :gem_information
+  def initialize(full_name)
```
Comment (Vadser) [inline 3469/1/html_parser.rb]: It is bad practice to return value from initialize, you should add another public method

File: 3469/1/program.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+class TopGems
+  include IoStream
+  def initialize(key)
```
Comment (Vadser) [inline 3469/1/program.rb]: It is bad practice to return value from initialize, you should add another public method

File: 3469/1/api.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    @gem_information
+  end
+
+  private
+
+  def url_of_site_from_github(name_of_gem)
+    @client.search_repositories(name_of_gem).items.first
+  end
+
+  def assign_the_parameters(gem_github_url)
```
Comment (Vadser) [inline 3469/1/api.rb]: gem_github_url is bad name for that

File: 3469/1/program.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    @key = key
+    @array_of_top_gems = []
+    @parameters = {}
+    @array_of_gems = []
+  end
+
+  def start_program
+    array_of_gems = information_from_file
+    read_parametrs_about_output
```
Comment (Vadser) [inline 3469/1/program.rb]: Please rename this method

File: 3469/1/program.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    start_search(array_of_gems)
+    sort_the_information
+    print_top(@array_of_top_gems)
+  end
+
+  private
+
+  def start_search(array_of_gems)
+    array_of_gems.map do |name_of_gem|
```
Comment (Vadser) [inline 3469/1/program.rb]: Rename

File: 3469/1/top_gems.yml [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+ - sinatra
+ - rspec
+ - devise
+ - rails
+ - colorize
+ - tmuxinator
+ - rake
+ - reek
+ - rubocop
+ - jquery-rails
```
Comment (Vadser) [inline 3469/1/top_gems.yml]: Add empty line

File: 3469/1/io_stream.rb [hunk 10]
```diff
@@ -0,0 +1,49 @@
+module IoStream
+  HEADER = %w[Name Used_by Watchers Stars Forks Contributors Issues].freeze
+
+  private
+
+  def information_from_file
+    read_parameters_about_file_name
+    begin
+     information = YAML.load_file(@parameters[:file])
+     @array_of_gems = information['gems']
```
Comment (Vadser) [inline 3469/1/io_stream.rb]: Rename array variable name

**Pull request merged**